src/Java.Interop.Tools.JavaTypeSystem/Adapters/JavaXmlApiImporter.cs

```diff
@@ -0,0 +1,370 @@
+						break;
+				}
+			}
+
+			return pkg;
+		}
+
+		public static JavaClassModel ParseClass (JavaPackage package, XElement element)
+		{
+			var model = new JavaClassModel (
+				javaPackage: package,
+				javaNestedName: element.XGetAttribute ("name"),
+				javaVisibility: element.XGetAttribute ("visibility"),
+				javaAbstract: element.XGetAttributeAsBool ("abstract"),
+				javaFinal: element.XGetAttributeAsBool ("final"),
+				javaBaseType: element.XGetAttribute ("extends"),
+				javaBaseTypeGeneric: element.XGetAttribute ("extends-generic-aware"),
+				javaDeprecated: element.XGetAttribute ("deprecated"),
+				javaStatic: element.XGetAttributeAsBool ("static"),
+				jniSignature: element.XGetAttribute ("jni-signature"),
+				baseTypeJni: element.XGetAttribute ("jni-extends")
+			);
+
+			if (element.XGetAttribute ("merge.SourceFile") is string source && source.HasValue ())
+				model.PropertyBag.Add ("merge.SourceFile", source);
+			if (element.XGetAttribute ("deprecated-since") is string dep && dep.HasValue ())
+				model.PropertyBag.Add ("deprecated-since", dep);
+
+			if (element.Element ("typeParameters") is XElement tp)
+				ParseTypeParameters (model.TypeParameters, tp);
+
+			foreach (var child in element.Elements ()) {
+				switch (child.Name.LocalName) {
+					case "constructor":
+						model.Constructors.Add (ParseConstructor (model, child));
+						break;
+					case "field":
+						model.Fields.Add (ParseField (model, child));
+						break;
+					case "implements":
+						model.Implements.Add (ParseImplements (child));
+						break;
+					case "method":
+						model.Methods.Add (ParseMethod (model, child));
+						break;
+				}
+			}
+
+			return model;
+		}
+
+		public static JavaInterfaceModel ParseInterface (JavaPackage package, XElement element)
+		{
+			var nested_name = element.XGetAttribute ("name");
+			var visibility = element.XGetAttribute ("visibility");
+			var deprecated = element.XGetAttribute ("deprecated");
+			var is_static = element.XGetAttribute ("static") == "true";
+			var jni_signature = element.XGetAttribute ("jni-signature");
+
+			var model = new JavaInterfaceModel (package, nested_name, visibility, deprecated, is_static, jni_signature);
+
+			if (element.XGetAttribute ("merge.SourceFile") is string source && source.HasValue ())
+				model.PropertyBag.Add ("merge.SourceFile", source);
+			if (element.XGetAttribute ("deprecated-since") is string dep && dep.HasValue ())
+				model.PropertyBag.Add ("deprecated-since", dep);
+
+			if (element.Element ("typeParameters") is XElement tp)
+				ParseTypeParameters (model.TypeParameters, tp);
+
+			foreach (var child in element.Elements ()) {
+				switch (child.Name.LocalName) {
+					case "field":
+						model.Fields.Add (ParseField (model, child));
+						break;
+					case "implements":
+						model.Implements.Add (ParseImplements (child));
+						break;
+					case "method":
+						if (child.XGetAttribute ("synthetic") != "true")
```

Comment:
Does this need to be "consistent" with: https://github.com/xamarin/java.interop/pull/849/files#diff-41afcd0fb56d03aabb52fc26b23822c138457f30e40940ae9d8b43e018e02690R199

(The "Here we skip most of the overriding methods of a virtual method, unless" comment block, above.)

Reply:
I think at this point we haven't figured out the rest of those conditions.  IE: we can't find the appropriate base method because it may not have been parsed yet.